### PR TITLE
Go rewrite services draft (Jack)

### PR DESCRIPTION
**Pull request by jr1221**

### CODE REVIEW

File: scylla-server-go/services/data_service/data_service.go [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+	"scylla-server/protobuf/serverdata"
+	"scylla-server/services/run_service"
+	"strconv"
+	"time"
+)
+
+func GetDataByDataTypeNameAndRunId(dataTypeName string, runId int) []db.DataModel {
+	client, ctx := prisma.PrismaClient()
+
+	dataType, err := client.DataType.FindUnique(db.DataType.Name.Equals(dataTypeName)).Exec(ctx)
```
Comment (Peyton-McKee) [inline scylla-server-go/services/data_service/data_service.go:20]: Do you get an error if it doesnt exist?

Comment (Peyton-McKee) [reply scylla-server-go/services/data_service/data_service.go:20]: If so then we want to error out not just console log

File: scylla-server-go/services/data_service/data_service.go [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+	log.Printf("Err %q\n", err.Error())
```
Comment (Peyton-McKee) [inline scylla-server-go/services/data_service/data_service.go]: Will this print regardless if theres an error or not

File: scylla-server-go/services/data_service/data_service.go [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+
+	var numbers []float64
+	for _, val := range serverData.Values {
+		if n, err := strconv.ParseFloat(val, 64); err == nil {
+			numbers = append(numbers, n)
+		}
+	}
+
+	created, err := client.Data.CreateOne(db.Data.DataType.Link(db.DataType.Name.Set(dataType.Name)), db.Data.Time.Set(time.UnixMilli(unixTime)), db.Data.Run.Link(db.Run.ID.Equals(run.ID)), db.Data.Values.Set(numbers)).Exec(ctx)
```
Comment (Peyton-McKee) [inline scylla-server-go/services/data_service/data_service.go:70]: We need a formatter

Comment (Peyton-McKee) [reply scylla-server-go/services/data_service/data_service.go:70]: And a linter if we dont have one yet

File: scylla-server-go/services/node_service/node_service.go [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+	data, err := client.Node.FindMany(db.Node.DataTypes.Every()).Exec(ctx)
+	log.Printf("Err %q\n", err.Error())
+
+	return data
+}
+
+func UpsertNode(nodeName string) *db.NodeModel {
+	client, ctx := prisma.PrismaClient()
+
+	node, err := client.Node.FindUnique(db.Node.Name.Equals(nodeName)).Exec(ctx)
```
Comment (Peyton-McKee) [inline scylla-server-go/services/node_service/node_service.go]: Is there no prisma.upsert statement?

File: scylla-server-go/services/run_service/run_service.go [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+	"scylla-server/prisma/db"
+	"time"
+)
+
+func GetAllRuns() []db.RunModel {
+	client, ctx := prisma.PrismaClient()
+	runs, err := client.Run.FindMany().Exec(ctx)
+	log.Printf("Err %q\n", err.Error())
+
+	// TODO: no mapping done
```
Comment (Peyton-McKee) [inline scylla-server-go/services/run_service/run_service.go]: We need mapping from prisma type to shared type

File: scylla-server-go/services/run_service/run_service.go [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+
+	return runs
+}
+
+func GetRunById(id int) *db.RunModel {
```
Comment (Peyton-McKee) [inline scylla-server-go/services/run_service/run_service.go]: These return types should all be shared types...

**Review by Peyton-McKee — CHANGES_REQUESTED** [review]
Overall looks good, looks like what i wouldve expected, def needs some cleaning up though with the types we should be returning and whatnot

File: scylla-server-go/services/node_service/node_service.go (hunk 4 above)
Comment (jr1221) [reply scylla-server-go/services/node_service/node_service.go]: There is but that's not what was used in the JS one.

**Comment by jr1221** [comment]
Fuck go.  Actually go pretty nice its just not as well supported by prisma, and it looks kinda mid.  Plus we get calypso and scylla on the same lang which would be epic.

**Pull request closed without merging**